caf-core-test target compilation failure with clang 10

The target `caf-core-test` currently fails to compile with clang 10, tested on macOS.

#### Steps to reproduce

1. `brew install llvm` (this installs clang 10)
2. Configure environment to use the bundled clang and libc++:
```
❯ typeset -pm {{LD,CPP,CXX}FLAGS,PATH,CMAKE_\*}
export -UT LDFLAGS ldflags=( -Wl,-rpath,/usr/local/opt/llvm/lib -L/usr/local/opt/llvm/lib ) ' '
export -UT CPPFLAGS cppflags=( '-isystem /usr/local/opt/llvm/include' ) ' '
export -UT CXXFLAGS cxxflags=( '-isystem /usr/local/opt/llvm/include/c++/v1' ) ' '
export -UT PATH path=( /usr/local/opt/ccache/libexec /usr/local/opt/llvm/bin /usr/local/bin /usr/local/sbin /usr/bin /bin /usr/sbin /sbin )
export -r CMAKE_EXPORT_COMPILE_COMMANDS=ON
export -r CMAKE_GENERATOR=Ninja
export -r CMAKE_BUILD_PARALLEL_LEVEL=12
```
3. Try to build current CAF master or even the latest release (0.17.4)
```
❯ ./configure
❯ cmake --build build --target caf-core-test
```

#### Error message

<details>
<summary>Click to expand</summary>

```
❯ cmake --build build --target caf-core-test --clean-first --parallel 1
[1/1 0.031s] Cleaning all built files...
Cleaning... 136 files.
[133/247 5.660s] Building CXX object libcaf_core/CMakeFiles/caf-core-test.dir/test/behavior.cpp.o
FAILED: libcaf_core/CMakeFiles/caf-core-test.dir/test/behavior.cpp.o
/usr/local/opt/ccache/libexec/clang++  -Dlibcaf_core_EXPORTS -I. -I../libcaf_core -I../libcaf_io -I../libcaf_test -I../libcaf_openssl -I../libcaf_core/test -isystem /usr/local/opt/llvm/include/c++/v1 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -fvisibility=hidden -fvisibility-inlines-hidden   -std=c++17 -ftemplate-depth=512 -ftemplate-backtrace-limit=0 -Wdocumentation -MD -MT libcaf_core/CMakeFiles/caf-core-test.dir/test/behavior.cpp.o -MF libcaf_core/CMakeFiles/caf-core-test.dir/test/behavior.cpp.o.d -o libcaf_core/CMakeFiles/caf-core-test.dir/test/behavior.cpp.o -c ../libcaf_core/test/behavior.cpp
In file included from ../libcaf_core/test/behavior.cpp:23:
In file included from ../libcaf_core/test/core-test.hpp:2:
../libcaf_test/caf/test/dsl.hpp:459:11: error: no viable overloaded '='
    dest_ = whom.ptr();
    ~~~~~ ^ ~~~~~~~~~~
../libcaf_core/caf/intrusive_ptr.hpp:120:18: note: candidate function not viable: no known conversion from 'caf::blocking_actor *' to 'caf::intrusive_ptr<caf::actor_control_block>::pointer' (aka 'caf::actor_control_block *') for 1st argument
  intrusive_ptr& operator=(pointer ptr) noexcept {
                 ^
../libcaf_core/caf/intrusive_ptr.hpp:125:18: note: candidate function not viable: no known conversion from 'caf::blocking_actor *' to 'caf::intrusive_ptr<caf::actor_control_block>' for 1st argument
  intrusive_ptr& operator=(intrusive_ptr other) noexcept {
                 ^
1 error generated.
ninja: build stopped: subcommand failed.
```

</details>

Let me know if you need any more info.

--- a/libcaf_test/caf/test/dsl.hpp
+++ b/libcaf_test/caf/test/dsl.hpp
@@ -456,7 +456,7 @@ public:
   }
 
   inject_clause& to(const caf::scoped_actor& whom) {
-    dest_ = whom.ptr();
+    dest_ = caf::actor_cast<caf::strong_actor_ptr>(whom.ptr());
     return *this;
   }
 

